_articles/aws_sts_storage_locations.md

```diff
@@ -0,0 +1,182 @@
+
+{% include important.html content="If a baseKey is **not** specified, the temporary AWS credentials vended by STS will give users access to the whole bucket. To prevent access to the whole bucket, enter a folder path in your bucket that all files in the storage location should go into as the baseKey. " %}
+
+##### Python
+
+```python
+# Set storage location
+import synapseclient
+import json
+syn = synapseclient.login()
+FOLDER = 'syn12345'
+
+destination = {'uploadType':'S3',
+               'stsEnabled':True,
+               'bucket':'nameofyourbucket',
+               'baseKey':'nameofyourbasekey',
+               'concreteType':'org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting'}
+destination = syn.restPOST('/storageLocation', body=json.dumps(destination))
+
+project_destination ={'concreteType': 'org.sagebionetworks.repo.model.project.UploadDestinationListSetting',
+                      'settingsType': 'upload'}
+project_destination['locations'] = [destination['storageLocationId']]
+project_destination['projectId'] = FOLDER
+
+project_destination = syn.restPOST('/projectSettings', body = json.dumps(project_destination))
+```
+
+##### R
+
+```r
+#set storage location
+library(synapser)
+library(rjson)
+synLogin()
+folderId <- 'syn12345'
+
+destination <- list(uploadType='S3',
+                    stsEnabled=TRUE,
+                    bucket='nameofyourbucket',
+                    baseKey='nameofyourbasekey',
+                    concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting')
+destination <- synRestPOST('/storageLocation', body=toJSON(destination))
+
+projectDestination <- list(concreteType='org.sagebionetworks.repo.model.project.UploadDestinationListSetting',
+                           settingsType='upload')
+projectDestination$locations <- list(destination$storageLocationId)
+projectDestination$projectId <- folderId
+
+projectDestination <- synRestPOST('/projectSettings', body=toJSON(projectDestination))
+```
+
+Once your STS Storage Location is set up on your Synapse folder, you can add files to Synapse in one of two ways:
+
+1. You can upload files through Synapse, either through [the Synapse website](https://www.synapse.org/) or through the client of your choice.
+2. If you plan to upload files directly to your S3 bucket, or if you already have files in your S3 bucket, you can add representations of those files to Synapse programmatically. Follow the steps at [Adding Files in Your S3 Bucket to Synapse](custom_storage_location.md#adding-files-in-your-s3-bucket-to-synapse).
+
+{% include note.html content="We recommend picking one of these methods and sticking to it and avoiding mixing methods." %}
```

Comment:
I updated the comment with further explanation: "Synapse automatically generates folders for files uploaded through the Synapse website or clients. Files added directly to S3 may not match this folder structure. We recommend against mixing these two methods of adding files to prevent confusion in the folder structure."